AttendanceSystem.Domain.Model/Session.cs

```diff
@@ -35,12 +35,32 @@ public Session(Course course, List<StudentId> students, DateTime startTime, Date
         {
             SessionId = Id,
             StudentId = s,
-            Record = Unknown
+            StudentAttendance = Unknown,
+            StudentSubmitted = null,
+            TeacherAttendance = Unknown,
+            TeacherSubmitted = null
         });
     }
 
-    public void SetAttendance(StudentId studentId, AttendanceKind kind)
-    {   
-        Register.Single(r => r.StudentId == studentId).Record = kind;
+    public void SetStudentAttendance(StudentId studentId, AttendanceKind attendance, DateTime submitted)
+    {
+        var record = Register.FirstOrDefault(r => r.StudentId == studentId);
+        if (record is null)
+        {
+            throw new ArgumentException("Student is not registered for this session.");
+        }
+        record.StudentAttendance = attendance;
+        record.StudentSubmitted = submitted;
+    }
+
+    public void SetTeacherAttendance(StudentId studentId, AttendanceKind attendance, DateTime submitted)
+    {
+        var record = Register.FirstOrDefault(r => r.StudentId == studentId);
+        if (record is null)
+        {
+            throw new ArgumentException("Student is not registered for this session.");
+        }
```

Comment:
I like the `??` better, but that is a matter of preference of course.
```suggestion
        var record = Register.FirstOrDefault(r => r.StudentId == studentId) ?? throw new ArgumentException("Student is not registered for this session.");
```

Same goes for the lines above.